EventFiltering/PWGHF/HFFilterHelpers.h

```diff
@@ -676,38 +734,48 @@ inline bool HfFilterHelper::isSelectedProton4Femto(const T1& track, const T2& tr
     return false; // use only global tracks
   }
 
-  float NSigmaTPC = track.tpcNSigmaPr();
-  float NSigmaTOF = track.tofNSigmaPr();
-  if (!forceTof && !track.hasTOF()) {
-    NSigmaTOF = 0.; // always accepted
+  // PID evaluation
+  float NSigmaTPC = (trackSpecies == kProtonForFemto) ? track.tpcNSigmaPr() : track.tpcNSigmaDe();
+  float NSigmaTOF = (trackSpecies == kProtonForFemto) ? track.tofNSigmaPr() : track.tofNSigmaDe();
+
+  // Handle TOF usage based on the pT threshold and requirements
+  if (pt <= ptThresholdPidStrategy) {
```

Comment:
maybe this we do only in the deuteron case, not to change the behaviour of the proton (not sure if we will keep the proton in the menu, but still)

Reply:
Ok, Then I separate the PID strategy for proton with deuteron, I keep the previous way for proton